antlr_ast/ast.py

```diff
@@ -19,188 +25,74 @@ def parse(grammar, text, start, strict=False, upper=True, error_listener=None):
     if strict:
         error_listener = StrictErrorListener()
 
-    if error_listener is not None:
+    if error_listener is not None and error_listener is not True:
         parser.removeErrorListeners()
         if error_listener:
             parser.addErrorListener(error_listener)
 
     return getattr(parser, start)()
 
 
-def dump_node(obj):
-    if isinstance(obj, AstNode):
+def process_tree(antlr_tree, Transformer=None, simplify=True):
+    tree = BaseAstVisitor().visit(antlr_tree)
+    if Transformer is not None:
+        tree = AliasVisitor(Transformer()).visit(tree)
+    if simplify:
+        tree = simplify_tree(tree, unpack_lists=False)
+    return tree
+
+
+# TODO
+#  duplicated in ast-viewer (also for Python)
+#  structure vs to_json()?
+def dump_node(node, node_class=AST):
+    if isinstance(node, node_class):
         fields = OrderedDict()
-        for name in obj._fields:
-            attr = getattr(obj, name, None)
+        for name in node._fields:
+            attr = getattr(node, name, None)
             if attr is None:
                 continue
-            elif isinstance(attr, AstNode):
+            elif isinstance(attr, node_class):
                 fields[name] = dump_node(attr)
             elif isinstance(attr, list):
                 fields[name] = [dump_node(x) for x in attr]
             else:
                 fields[name] = attr
-        return {"type": obj.__class__.__name__, "data": fields}
-    elif isinstance(obj, list):
-        return [dump_node(x) for x in obj]
+        return {"type": node.__class__.__name__, "data": fields}
+    elif isinstance(node, list):
+        return [dump_node(x) for x in node]
     else:
-        return obj
+        return node
 
 
 FieldSpec = namedtuple("FieldSpec", ["name", "origin"])
 
 
 def parse_field_spec(spec):
-    # parse mapping for -> and indices [] -----
-    origin, *name = spec.split("->")
-    name = origin if not name else name[0]
+    # parse mapping for = and .  # old: and indices [] -----
+    name, *origin = spec.split("=")
```

Comment:
Do we want to trim spaces? E.g. what happens for `a = b`?

Reply:
Currently it would fail with spaces (so the depending libraries don't use spaces), but I added trimming after splitting.